diff -ur 1.83.0/overlay/BUILD.bazel b/BUILD.bazel
--- a/BUILD.bazel	2025-03-28 10:37:27.434605482 -0700
+++ b/BUILD.bazel	2025-03-28 17:41:06.659641196 -0700
@@ -1,15 +1,25 @@
+load("@bazel_skylib//rules:write_file.bzl", "write_file")
 load("@rules_cc//cc:defs.bzl", "cc_library")
 
 package(default_visibility = ["//visibility:public"])
 
+write_file(
+    name = "src",
+    out = "boost.beast.src.cpp",
+    content = ["#include <boost/beast/src.hpp>"],
+    visibility = ["//visibility:private"],
+)
+
 cc_library(
     name = "boost.beast",
+    srcs = [":src"] + ["include/boost/beast/src.hpp"],
     hdrs = glob(
         [
             "include/**/*.hpp",
             "include/**/*.ipp",
         ],
         exclude = [
+            "include/boost/beast/src.hpp",
             "include/boost/beast/_experimental/http/impl/icy_stream.hpp",
             "include/boost/beast/_experimental/test/detail/stream_state.ipp",
             "include/boost/beast/_experimental/test/impl/error.hpp",
diff -ur 1.83.0/overlay/MODULE.bazel b/MODULE.bazel
--- a/MODULE.bazel	2025-03-28 10:39:31.401630035 -0700
+++ b/MODULE.bazel	2025-03-28 17:36:06.232581653 -0700
@@ -5,6 +5,7 @@
     compatibility_level = 108300,
 )
 
+bazel_dep(name = "bazel_skylib", version = "1.7.1")
 bazel_dep(name = "boost.asio", version = "1.83.0.bcr.1")
 bazel_dep(name = "boost.assert", version = "1.83.0.bcr.1")
 bazel_dep(name = "boost.bind", version = "1.83.0.bcr.1")
