diff --git a/include/ord/OpenRoad.hh b/include/ord/OpenRoad.hh
index 80d28c4dc..2325b3437 100644
--- a/include/ord/OpenRoad.hh
+++ b/include/ord/OpenRoad.hh
@@ -150,6 +150,10 @@ namespace dft {
 class Dft;
 }
 
+namespace tool{
+class Tool;
+}
+
 namespace ord {
 
 using std::string;
@@ -195,6 +199,7 @@ class OpenRoad
   dst::Distributed* getDistributed() { return distributer_; }
   stt::SteinerTreeBuilder* getSteinerTreeBuilder() { return stt_builder_; }
   dft::Dft* getDft() { return dft_; }
+  tool::Tool* getTool() { return tool_; }
 
   // Return the bounding box of the db rows.
   odb::Rect getCore();
@@ -276,6 +281,7 @@ class OpenRoad
   dst::Distributed* distributer_ = nullptr;
   stt::SteinerTreeBuilder* stt_builder_ = nullptr;
   dft::Dft* dft_ = nullptr;
+  tool::Tool* tool_ = nullptr;
 
   std::set<OpenRoadObserver*> observers_;
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 3fbb95f74..6a2759364 100755
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -270,6 +270,7 @@ add_subdirectory(utl)
 add_subdirectory(dst)
 add_subdirectory(pdn)
 add_subdirectory(dft)
+add_subdirectory(tool)
 
 ################################################################
 
@@ -331,6 +332,7 @@ target_link_libraries(openroad
   utl
   pdn
   dft
+  tool
   ${ABC_LIBRARY}
   ${TCL_LIBRARY}
   ${CMAKE_THREAD_LIBS_INIT}
diff --git a/src/OpenRoad.cc b/src/OpenRoad.cc
index 1cf9ff0f7..4a5b31732 100644
--- a/src/OpenRoad.cc
+++ b/src/OpenRoad.cc
@@ -87,6 +87,7 @@
 #include "triton_route/MakeTritonRoute.h"
 #include "utl/Logger.h"
 #include "utl/MakeLogger.h"
+#include "tool/MakeTool.hh"
 
 namespace sta {
 extern const char* openroad_swig_tcl_inits[];
@@ -154,6 +155,7 @@ OpenRoad::~OpenRoad()
   deleteDistributed(distributer_);
   deleteSteinerTreeBuilder(stt_builder_);
   dft::deleteDft(dft_);
+  deleteTool(tool_);
   delete logger_;
 }
 
@@ -212,6 +214,7 @@ void OpenRoad::init(Tcl_Interp* tcl_interp)
   distributer_ = makeDistributed();
   stt_builder_ = makeSteinerTreeBuilder();
   dft_ = dft::makeDft();
+  tool_ = makeTool();
 
   // Init components.
   Openroad_swig_Init(tcl_interp);
@@ -252,6 +255,7 @@ void OpenRoad::init(Tcl_Interp* tcl_interp)
   initDistributed(this);
   initSteinerTreeBuilder(this);
   dft::initDft(this);
+  initTool(this);
 
   // Import exported commands to global namespace.
   Tcl_Eval(tcl_interp, "sta::define_sta_cmds");
